frontend/src/app/active-sessions/upload-dialog/upload-dialog.component.ts

```diff
@@ -246,6 +248,14 @@ export class UploadDialogComponent implements OnInit, OnDestroy {
       });
   }
 
+  download(filename: string) {
+    this.loadService
+      .download(this.session.id, filename)
+      .subscribe((response: Blob) => {
+        saveAs(response, 'download.zip');
```

Comment:
It would be nice to see a progress of the download. I tried it with our largest models, I clicked the button and for about one minute, nothing happened. After one minute, the download was just there. 

One workaround could be to use just a href tag: https://stackoverflow.com/a/61190093, but this would only capture the download progress between the backend and the frontend (not between container and backend). In the end we are at the point of implementing async again.

Reply:
I added some fancy progress bars (see screenshot above)